### DISCUSSION TITLE
What's the explicit difference between this and vim-fugitive?

### DISCUSSION THREAD

**Discussion by Ciel-MC** [NeogitOrg/neogit#539] [Q&A] [answered]
I randomly came up on this recently (somehow didn't see it when looking for git plugins initially), and this looks quite similar to fugitive, is there any explicit design difference or whatnot that made this happen? Or are there similar principals but in just Lua?

**Comment by CKolkey** [comment]
Hi! This plugin doesn't take any inspiration from fugitive, but I can tell you why they feel similar: fugitive took inspiration from the same place this does! https://github.com/tpope/vim-fugitive/issues/569

What we're doing here is working towards building [magit](https://magit.vc), an emacs plugin, in lua, for neovim. Any similarities to fugitive are purely coincidental :)

**Reply by Ciel-MC** [reply]
Aha! So they both have Magit root, I never saw fugitive mention that. So it's just the same root, but nothing is made purposefully the same, got it. Gonna try it out some time, thanks a lot :D

**Reply by CKolkey** [reply]
Exactly :) I think it would be fair to say that Fugitive is very much its own thing, and drew inspiration from Magit, whereas this project is an explicit copy. 

We're in the process of a rather major upgrade right now - Keep an eye on when the CKolkey branch hits master 🥳

**Reply by ten3roberts** [reply, marked as answer]
The CKolkey fork has now arrived in master.

And you are correct, Fugitive is its own thing which serves to be a Magit inspired solution for Git workflow inside Vim.

Fugitive uses a more Vim idiomatic workflow, such as relying on Ex commands and as such works more like a smart terminal which you feed commands.

Neogit is perhaps closer to Magit and also serves to bring a Git workflow to Neovim. However, Neogit aims to achive a much more interactive interface with popups, navigation, keybindings and selection screens. An example which neatly illustrates this is the rebasing workflow. 

Fugitive provides a status like buffer which displays the current git state and in-progress rebases and staged/unstaged files. However, you are expected to use the vim approach and enter the rebase command exactly the way you want it using `:Git ...` where fugitive will sprinkle QOL ontop.

Neogit will allow you to press `r` and get a full interactive popup for rebasing and all the options it entails, as well as providing a selection buffer of all found branches for the rebase target branch.

In short, the long term goal of Neogit is to provide a keyboard oriented interactive workflow for your daily (and weekly) git needs without writing raw git commands
